libraries/SITL/SIM_GPS.cpp

```diff
@@ -698,7 +698,7 @@ void GPS::update_sbp(const struct gps_data *d)
     pos.height = d->altitude;
     pos.h_accuracy = _sitl->gps_accuracy[instance]*1000;
     pos.v_accuracy = _sitl->gps_accuracy[instance]*1000;
-    pos.n_sats = _sitl->gps_numsats[instance];
+    pos.n_sats = d->have_lock?_sitl->gps_numsats[instance]:3;
```

Comment:
```suggestion
    pos.n_sats = d->have_lock ? _sitl->gps_numsats[instance] : 3;
```

similarly elsewhere

Reply:
Fixed in 750adf4e86f12776878d1d99ba3177faf1edeba8 for all the occurences.